File: opcua-webservice-spring/core/src/main/java/org/opcfoundation/webserver/addressspace/nodes/builtin/UaVariableTypes.java
package org.opcfoundation.webserver.addressspace.nodes.builtin;

import org.eclipse.milo.opcua.stack.core.NodeIds;
import org.eclipse.milo.opcua.stack.core.types.builtin.LocalizedText;
import org.opcfoundation.webserver.addressspace.nodes.UaVariableType;

public class UaVariableTypes {
    public final static UaVariableType BaseVariableType = new UaVariableType(NodeIds.BaseVariableType,"BaseVariableType", new LocalizedText("BaseVariableType"), true, NodeIds.BaseDataType, -2);
    public final static UaVariableType BaseDataVariableType = new UaVariableType(NodeIds.BaseDataVariableType,"BaseDataVariableType", new LocalizedText("BaseDataVariableType"), false, NodeIds.BaseDataType,-2);
    public final static UaVariableType PropertyType = new UaVariableType(NodeIds.PropertyType,"PropertyType", new LocalizedText("PropertyType"), false, NodeIds.BaseDataType,-2);

    public final static UaVariableType DataItemType = new UaVariableType(NodeIds.DataItemType,"DataItemType", new LocalizedText("DataItemType"), false, NodeIds.BaseDataVariableType,-2);
    public final static UaVariableType BaseAnalogType = new UaVariableType(NodeIds.BaseAnalogType,"BaseAnalogType", new LocalizedText("BaseAnalogType"), false, NodeIds.Number,-2);
}

File: opcua-webservice-node/source/src/server/addressspace/nodes/builtin/UaVariableTypes.ts
import { DataTypeIds, UaLocalizedText, UaNodeId, VariableTypeIds } from "opcua-webapi-ts";
import { UaVariableType } from "../UaVariableType";

export class UaVariableTypes {
    static readonly BaseVariableType = new UaVariableType(UaNodeId.from(VariableTypeIds.BaseVariableType), "BaseVariableType", new UaLocalizedText("BaseVariableType"), true, UaNodeId.from(DataTypeIds.BaseDataType), -1);
    static readonly BaseDataVariableType = new UaVariableType(UaNodeId.from(VariableTypeIds.BaseDataVariableType), "BaseDataVariableType", new UaLocalizedText("BaseDataVariableType"), false, UaNodeId.from(DataTypeIds.BaseDataType), -1);
    static readonly PropertyType = new UaVariableType(UaNodeId.from(VariableTypeIds.PropertyType), "PropertyType", new UaLocalizedText("PropertyType"), false, UaNodeId.from(DataTypeIds.BaseDataType), -1);

    static readonly DataItemType = new UaVariableType(UaNodeId.from(VariableTypeIds.DataItemType), "DataItemType", new UaLocalizedText("DataItemType"), false, UaNodeId.from(DataTypeIds.BaseDataType), -1);
    static readonly BaseAnalogItemType = new UaVariableType(UaNodeId.from(VariableTypeIds.BaseAnalogType), "BaseAnalogType", new UaLocalizedText("BaseAnalogType"), false, UaNodeId.from(DataTypeIds.Number), -1);
}